server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -168,7 +170,7 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
       intervalsToSkipCompaction.computeIfAbsent(status.getDataSource(), k -> new ArrayList<>())
                                .add(interval);
 
-      busyCompactionTaskSlots += findMaxNumTaskSlotsUsedByOneCompactionTask(
+      busyCompactionTaskSlots += findMaxNumTaskSlotsUsedByOneNativeCompactionTask(
```

Comment:
Why do we assume that the currently running one is a native compaction task? We should check if the context of this task has the MSQ tasks parameter set.

Reply:
Thanks for spotting this. Corrected this now to retrieve the task's engine-type and calculate busy slots based on that.